File: python/templative/lib/distribute/playground/playgroundSettingsManagement.py
from os import path
from aiofile import AIOFile
    
async def writePlaygroundFile(outputPath):
    playgroundFileLocation = path.join("./", ".playground")
    async with AIOFile(playgroundFileLocation, mode="w") as playground:
        await playground.write(outputPath)

async def getPlaygroundDirectory(inputedPlaygroundDirectory):
    if inputedPlaygroundDirectory != None:
        return inputedPlaygroundDirectory
    
    playgroundFileLocation = "./.playground"
    if not path.exists(playgroundFileLocation):
        return None
    
    async with AIOFile(playgroundFileLocation, mode="r") as playground:
        return await playground.read()

File: src/main/templative/lib/distribute/playground/playgroundSettingsManagement.js
const fs = require('fs');
const fsPromises = fs.promises;
const path = require('path');

async function writePlaygroundFile(outputPath) {
    const playgroundFileLocation = path.join("./", ".playground");
    await fsPromises.writeFile(playgroundFileLocation, outputPath);
}

async function getPlaygroundDirectory(inputedPlaygroundDirectory) {
    if (inputedPlaygroundDirectory !== undefined) {
        return inputedPlaygroundDirectory;
    }
    
    const playgroundFileLocation = "./.playground";
    try {
        return await fsPromises.readFile(playgroundFileLocation, 'utf8');
    } catch (error) {
        if (error.code !== 'ENOENT') {
            throw error;
        }
        return undefined;
    }
}

module.exports = {
    writePlaygroundFile,
    getPlaygroundDirectory
};